clientcompat/internal/clientcompat/clientcompat.twirp.go

```diff
@@ -180,10 +180,10 @@ func (s *compatServiceServer) ServeHTTP(resp http.ResponseWriter, req *http.Requ
 	}
 
 	switch req.URL.Path {
-	case "/twirp/twirp.clientcompat.CompatService/Method":
+	case "/twirp.clientcompat.CompatService/Method":
 		s.serveMethod(ctx, resp, req)
 		return
-	case "/twirp/twirp.clientcompat.CompatService/NoopMethod":
+	case "/twirp.clientcompat.CompatService/NoopMethod":
```

Comment:
Just removing the prefix is not enough. The base-URL now can have path components. For example, to upgrade this service while keeping compatibility, before was mounted on "http://localhost:8000" but now it should be mounted on "http://localhost:8000/twirp". The routing should check for `req.URL.Path` suffix, not exact match.

Reply:
Of course, I am not sure how I forgot that in this first pass. Will fix.

Reply:
Oh, I _do_ remember how: we don't want a suffix match; we expect users to use [`net/http.StripPrefix`](https://golang.org/pkg/net/http/#StripPrefix) when setting up their server if they want to use a prefix.

Reply:
Ah ok, but then we will have to be careful documenting this in the docs. Please make sure that the Haberdasher example shows how to properly mount the service in the "/twirp" prefix, with a NOTE for older versions.